### PR TITLE
fix bug in 'develop a new chaos' in website

### LINKED ISSUE #985: website: change needed in `Develop a New Chaos`
## Bug Report

**What version of Kubernetes are you using?**
v1.18.0

There are two places need to change:
1. In [Add the chaos object in controller](https://chaos-mesh.org/docs/development_guides/develop_a_new_chaos/#add-the-chaos-object-in-controller), no need to change controller-manager-rbac.yaml anymore, since in #872 
> chaos-mesh controller could work without permission of cluster scope
2. In [Make the Docker image](https://chaos-mesh.org/docs/development_guides/develop_a_new_chaos/#make-the-docker-image), we need to do `make generate` before `make`, since otherwise the error would shows like 
```
#18 1.579 api/v1alpha1/helloworldchaos_types.go:38:25: cannot use &HelloWorldChaos literal (type *HelloWorldChaos) as type runtime.Object in argument to SchemeBuilder.Register:
#18 1.579 *HelloWorldChaos does not implement runtime.Object (missing DeepCopyObject method)
```

### PR DESCRIPTION
**Pull request by Yiyiyimu**
Signed-off-by: yiyiyimu <wosoyoung@gmail.com>

### What problem does this PR solve?
fix #985 

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by zhouqiang-cl** [comment]
@WangXiangUSTC PTAL

File: website/docs/development_guides/dev_hello_world.md [hunk 1]
```diff
@@ -192,7 +180,7 @@ Now take the following steps to run chaos:
 2. Install Chaos Mesh:
 
     ```bash
-    helm install helm/chaos-mesh --name=chaos-mesh --namespace=chaos-testing --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock
+    helm install chaos-mesh helm/chaos-mesh --namespace=chaos-testing --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock
```
Comment (WangXiangUSTC) [inline website/docs/development_guides/dev_hello_world.md:183]: how about providing two commands for helm v3 and helm v2?

File: website/docs/development_guides/dev_hello_world.md [hunk 2]
```diff
@@ -144,13 +130,15 @@ metadata:
...
 ```bash
+make generate
 make
 make docker-push
 ```
 
 > **Note:**
 >
 > The default `DOCKER_REGISTRY` is `localhost:5000`, which is preset in `hack/kind-cluster-build.sh`. You can overwrite it to any registry to which you have access permission.
+> To run a local registry with KinD, you could follow the [official doc](https://kind.sigs.k8s.io/docs/user/local-registry/).
```
Comment (WangXiangUSTC) [inline website/docs/development_guides/dev_hello_world.md:141]: https://github.com/chaos-mesh/chaos-mesh/pull/996
in this pr will run the registry automatically, maybe this modified is not necessary.

File: website/docs/development_guides/dev_hello_world.md (hunk 1 above)
Comment (Yiyiyimu) [reply website/docs/development_guides/dev_hello_world.md:183]: Sure that would be better. thx!

File: website/docs/development_guides/dev_hello_world.md (hunk 2 above)
Comment (Yiyiyimu) [reply website/docs/development_guides/dev_hello_world.md:141]: That would be great. Would fix it later

File: website/docs/development_guides/dev_hello_world.md [hunk 3]
```diff
@@ -70,20 +70,6 @@ In Chaos Mesh, all chaos types are managed by the controller manager. To add a n
     }
     ```
 
-> **Note:**
```
Comment (WangXiangUSTC) [inline website/docs/development_guides/dev_hello_world.md:73]: why remove these note

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

**Comment by YangKeao** [comment]
:man_facepalming: I have rewritten this document according to the `chaos-build` code generator. Can you help me check whether it's correct?

File: website/docs/development_guides/dev_hello_world.md (hunk 3 above)
Comment (Yiyiyimu) [reply website/docs/development_guides/dev_hello_world.md:73]: since in #872 
> chaos-mesh controller could work without permission of cluster scope

**Comment by Yiyiyimu** [comment]
> 🤦‍♂️ I have rewritten this document according to the `chaos-build` code generator. Can you help me check whether it's correct?

Sure 😃

**Comment by Yiyiyimu** [comment]
> 🤦‍♂️ I have rewritten this document according to the `chaos-build` code generator. Can you help me check whether it's correct?

Didn't occur error but it seems helloworldchaos is not triggered in the end. I [uploaded the code](https://github.com/Yiyiyimu/chaos-mesh/tree/test) for reference, and I'll look into it tonight.
image

**Comment by Yiyiyimu** [comment]
https://chaos-mesh.org/docs/next/development_guides/develop_a_new_chaos/#register-the-handler-for-this-chaos-object

This step should create a file named types.go, but rename still couldn't fix the problem

**Comment by YangKeao** [comment]
@Yiyiyimu I don't think rename will make a difference (and I prefer `endpoint.go`, because I know that `types.go` is a bad habbit of myself :crying_cat_face:  ). The expected log should contain:

```
2020-10-14T06:06:33.508Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "helloworldchaos", "source": "kind source: /, Kind="}
```

**Comment by YangKeao** [comment]
@Yiyiyimu Your code works well on my cluster :crying_cat_face: . Can you check whether you have rebuilt your image and the image hash is the same.

**Comment by Yiyiyimu** [comment]
> @Yiyiyimu Your code works well on my cluster 😿 . Can you check whether you have rebuilt your image and the image hash is the same.

Yeah I just found it works... maybe there is something left for the non-helloworld version, and a full-reboot remove it... 
Anyway it also works on my side

**Review by cwen0 — APPROVED** [review]
LGTM

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**